winpr/libwinpr/clipboard/dirent.c

```diff
@@ -0,0 +1,85 @@
+
+#include <stdio.h>
+#include "dirent.h"
+
+#include "../log.h"
+#define TAG WINPR_TAG("clipboard.synthetic.file")
+
+DIR* opendir(const char* name)
+{
+	DIR* dir;
+	WIN32_FIND_DATA FindData;
+	char namebuf[512];
+
+	sprintf(namebuf, "%s\\*.*", name);
+
+	HANDLE hFind = FindFirstFileA(namebuf, &FindData);
```

Comment:
Better use `FindFirstFileW` on windows, or unicode file names will not work.

Reply:
Yes. But the input parameter is const char* local_name

static BOOL add_directory_contents_to_list(wClipboard* clipboard, const char* local_name,
                                           const WCHAR* remote_name, wArrayList* files)

Need to modify all functions?

Reply:
@KangLin no, expect the input to use `utf-8`.

Reply:
OK. It has been modified. but it is only work in utf-8.

My OS:
 Microsoft Windows [Version 10.0.19044.2006]
 Simplified Chinese Windows is GB2313 - CodePage 936

It is not work.

Reply:
After testing, it turns out that using FindFirstFileA is correct. Pelease see TestDirent.c

Reply:
@akallabeth  My understanding is that multi-byte parameters are char*, and wide-byte(UNICODE) parameters are wchar_t*.
FindFirstFileA handles multi-byte parameters (char*), include ASNI and UTF-8
FindFirstFileW handles wide-byte(UNICODE) parameters (wchar_t*). include UTF-16、GB2313 is 2byte, UTF-32 is 4byte etc.
Since the current function [DIR* opendir(const char* name), expect the input to use utf-8.] is passing a mult-ibyte parameter(char*), I think it is correct to use FindFirstFileA. 
If the function [DIR* opendir(const wchar_t* name), expect the input to use UTF-16、GB2313.] is passing a wide-byte parameter(wchar_t*), then use FindFirstFileW.

Reply:
@KangLin No, `FindFirstFileA` is `multi-byte`, yes, but not `utf-8` but current code page. we´ve fixed similar issues for freerdp on windows on multiple occasions.